### PR TITLE
docs: update LTS versions and expand on policy

### PR DESCRIPTION
**Pull request by nyonson**
Updating with the recent call to drop LTS on `0.31.x`. Also refactored the policy doc with my interpretation which may bring some clarity to others (or myself!).

### CODE REVIEW

File: docs/supported-versions.md [hunk 1]
```diff
@@ -1,58 +1,22 @@
...
 
-## Bitcoin LTS versions
+Because the `bitcoin` crate is an umbrella across the rest of the crates in the workspace, the oldest LTS version's minimum dependencies determine the LTS versions of the other crates.
 
-- `v0.30` (Security and bug fixes only)
-- `v0.31` (Security and bug fixes only)
-- `v0.32`
-
-`v0.32` is the version most of the eccosystem is on. It is actively maintained. This means:
+LTS versions are actively maintained and receive updates like the following.
```
Comment (tcharding) [inline docs/supported-versions.md:7]: This is a change in policy (that I don't think we should commit to). Currently we only guarantee to backport features to the release before master. I.e yes to 0.32 but no to 0.31 but if I'm reading this text correctly it say we will backport features to the `0.32.10` series too. I rekon that should be security only. Defer to @apoelstra though.

File: docs/supported-versions.md [hunk 2]
```diff
@@ -1,58 +1,22 @@
...
-  -> `units-1.0.0`
-  -> `consensus-encoding 1.0.0`
+feature with the explicit aim of then backporting to LTS branches.
+
+## `bitcoin` LTS Versions
+
+- `v0.32.100+` -- MSRV of `1.60.0` and the optional `encoding` feature.
+- `v0.32.10+` -- MSRV of `1.56.1`.
+
+As of 2026-07-07, `v0.32` is the version most of the ecosystem is on and is actively maintained. Due to its longevity, we are in the unconventional spot of maintaining two branches of `v0.32` in an effort to make it easier for user to migrate to the stabilized crates.
```
Comment (tcharding) [inline docs/supported-versions.md]:
```suggestion
As of 2026-07-07, `v0.32` is the version most of the ecosystem is on and is actively maintained. Due to its longevity, we are in the unconventional spot of maintaining two branches of `v0.32` in an effort to make it easier for users to migrate to the stabilized crates.
```

File: docs/supported-versions.md (hunk 1 above)
Comment (nyonson) [reply docs/supported-versions.md:7]: Ah, ok, yea this was me misinterpreting the original policy. FWIW I think it makes sense to try and have the same policy for both LTS versions, but with a caveat that if a backport is too hard for either it might get cut...which I think rule 1 covers?

**Comment by nyonson** [comment]
[d704b49](https://github.com/rust-bitcoin/rust-bitcoin/commit/d704b4947a0db8cb15d8236d9f746fbe384c21ef): fix typo

File: docs/supported-versions.md (hunk 1 above)
Comment (apoelstra) [reply docs/supported-versions.md:7]: I think it's fine. "actively maintained" to me means that we'll make CI pass. "Receives updates" means "if it seems important and we feel like it" which is basically what the text says.

**Comment by apoelstra** [comment]
An interesting test case for this policy is -- #6317 needs a backport to 0.32.x. Obviously we will backport it to 0.32.100. Will we backport it to 0.32.10? It's a minor edge case where we decode some kinds of malformed `sendcmpt` messages.

**Review by apoelstra — APPROVED** [review]
ACK d704b4947a0db8cb15d8236d9f746fbe384c21ef; successfully ran local tests; short and sweet

File: docs/supported-versions.md (hunk 1 above)
Comment (tcharding) [reply docs/supported-versions.md:7]: > if it seems important and we feel like it

My type of work.

**Review by tcharding — APPROVED** [review]
ACK d704b4947a0db8cb15d8236d9f746fbe384c21ef
